README.md

```diff
@@ -8,96 +8,56 @@ annotation-based dependency injection container. Have a look at the [disco-demos
 
 ```
 composer.phar require bitexpert/disco
 ```
 
 ## Usage
 
-First, you need to define a configuration class. The class needs to be marked with `@Configuration` annotation:
+To instanciate Disco use the following code in your bootstrapping logic.
+Create an instance of the `\bitExpert\Disco\AnnotationBeanFactory` and
+register the instance with the `\bitExpert\Disco\BeanFactoryRegistry`.
+The second step is important as Disco needs to grab the active container
+instance in a few locations where it does not have access to the
+container instance itself.
 
 ```php
 <?php
 
-use bitExpert\Disco\Annotations\Configuration;
-
-/**
- * @Configuration
- */
-class MyConfiguration
-{
-}
-```
-
-Add a public or protected method for each instance you want Disco to manage for you. Instances managed by a public
-method are accessible via the `has()` and `get()` methods of the `\bitExpert\Disco\AnnotationBeanFactory` class. Any
-instance managed by a protected method is seen as a local dependency, and is just accessible "inside" the container's
-configuration class.  This is useful for dependencies like database connectors, which you most likely do not want to
-expose to the outside world.
-
-Each method needs to be marked with `@Bean` annotation as well as define the [return type declaration](http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration)
-(which is mainly needed for the lazy proxy magic).
-
-```php
-<?php
-
-use bitExpert\Disco\Annotations\Bean;
-use bitExpert\Disco\Annotations\Configuration;
-use bitExpert\Disco\Helper\SampleService;
-
-/**
- * @Configuration
- */
-class MyConfiguration
-{
-    /**
-     * @Bean
-     */
-    public function mySampleService() : SampleService
-    {
-        return new SampleService();
-    }
-}
+$beanFactory = new \bitExpert\Disco\AnnotationBeanFactory(MyConfiguration::class);
+\bitExpert\Disco\BeanFactoryRegistry::register($beanFactory);
 ```
 
-The `@Bean` annotation comes with a few configuration settings for influencing the lifetime of the instance. You can
-define the `scope` (which is either the string `request` or `session`), define a boolean value for `lazy`, or define
-a boolean value for `singleton`. The following configuration creates a lazy-loaded, singleton, request-aware instance:
+Next up you need to create a configuration class `MyConfiguration`. The
+class needs to be marked with a @Configuration annotation.
 
 ```php
 <?php
 
-use bitExpert\Disco\Annotations\Bean;
 use bitExpert\Disco\Annotations\Configuration;
-use bitExpert\Disco\Helper\SampleService;
 
 /**
  * @Configuration
  */
 class MyConfiguration
 {
-    /**
-     * @Bean({"singleton"=true, "lazy"=true, "scope"="request"})
-     */
-    public function mySampleService() : SampleService
-    {
-        return new SampleService();
-    }
 }
 ```
 
-To inject a dependency, simply call the respective method:
+To declare an configuration entry, simply add a method to your
```

Comment:
Should be "a configuration entity" instead of "an configuration entity. [Here's the rule](http://www.quickanddirtytips.com/education/grammar/a-versus-an) I base this statement on.